src/tracks/playabletrack/wavetrack/ui/ChangeClipSpeedDialog.cpp

```diff
@@ -11,106 +11,237 @@
 #include "ChangeClipSpeedDialog.h"
 
 #include <wx/button.h>
+#include <wx/layout.h>
+#include <wx/spinctrl.h>
 #include <wx/textctrl.h>
 
 #include "ShuttleGui.h"
+#include "SpinControl.h"
 #include "wxWidgetsWindowPlacement.h"
 
+namespace
+{
+template <typename ReturnType, typename... Args> class ScopedSizer
+{
+public:
+   ScopedSizer(
+      ShuttleGui& s, ReturnType (ShuttleGui::*startFunc)(Args...),
+      void (ShuttleGui::*endFunc)(), Args&&... args)
+       : mShuttleGui(s)
+       , mStartFunction(startFunc)
+       , mEndFunction(endFunc)
+   {
+      (mShuttleGui.*mStartFunction)(std::forward<Args>(args)...);
+   }
+
+   ~ScopedSizer()
+   {
+      (mShuttleGui.*mEndFunction)();
+   }
+
+private:
+   ShuttleGui& mShuttleGui;
+   ReturnType (ShuttleGui::*mStartFunction)(Args...);
+   void (ShuttleGui::*mEndFunction)();
+};
+
+class ScopedHorizontalLay : public ScopedSizer<void, int, int>
+{
+public:
+   ScopedHorizontalLay(
+      ShuttleGui& s, int PositionFlags = wxALIGN_CENTRE, int iProp = 1)
+       : ScopedSizer<void, int, int>(
+            s, &ShuttleGui::StartHorizontalLay, &ShuttleGui::EndHorizontalLay,
+            std::move(PositionFlags), std::move(iProp))
+   {
+   }
+};
+
+class ScopedInvisiblePanel : public ScopedSizer<wxPanel*, int>
+{
+public:
+   ScopedInvisiblePanel(ShuttleGui& s, int border = 0)
+       : ScopedSizer<wxPanel*, int>(
+            s, &ShuttleGui::StartInvisiblePanel, &ShuttleGui::EndInvisiblePanel,
+            std::move(border))
+   {
+   }
+};
+
+class ScopedVerticalLay : public ScopedSizer<void, int>
+{
+public:
+   ScopedVerticalLay(ShuttleGui& s, int iProp = 1)
+       : ScopedSizer<void, int>(
+            s, &ShuttleGui::StartVerticalLay, &ShuttleGui::EndVerticalLay,
+            std::move(iProp))
+   {
+   }
+};
+} // namespace
```

Comment:
Those classes look nice, but I'd rather you just follow conventional use of ShuttleGui for bracketing things.

Reply:
It helped me when I had to change things around, and I think it would also have helped me when I looked at this source file the first time and was trying to figure out how it all worked.